tf2_sensor_msgs/test/test_tf2_sensor_msgs.py

```diff
@@ -29,61 +29,186 @@
-        self.point_cloud_in.width = 2
-        self.point_cloud_in.row_step = \
-            self.point_cloud_in.point_step * self.point_cloud_in.width
-
-        points = [1, 2, 0, 10, 20, 30]
-        self.point_cloud_in.data = struct.pack('%sf' % len(points), *points)
-
-        self.transform_translate_xyz_300 = TransformStamped()
-        self.transform_translate_xyz_300.transform.translation.x = 300
-        self.transform_translate_xyz_300.transform.translation.y = 300
-        self.transform_translate_xyz_300.transform.translation.z = 300
-        # no rotation so we only set w
-        self.transform_translate_xyz_300.transform.rotation.w = 1
+        self.points = np.array([[1, 2, 0], [10, 20, 30]], dtype=np.float32)
 
-        assert(list(point_cloud2.read_points(self.point_cloud_in)) ==
-               [(1.0, 2.0, 0.0), (10.0, 20.0, 30.0)])
+        self.point_cloud_in = create_cloud_xyz32(
+            Header(frame_id="test"),
+            self.points)
 
     def test_simple_transform(self):
-        # deepcopy is not required here because we have a str for now
+        k = 300.0
+        transform_translate_xyz = TransformStamped()
+        transform_translate_xyz.transform.translation.x = k
+        transform_translate_xyz.transform.translation.y = k
+        transform_translate_xyz.transform.translation.z = k
+        # no rotation so we only set w
+        transform_translate_xyz.transform.rotation.w = 1.0
+
+        # Check if point cloud contains our points
+        self.assertTrue(np.allclose(
+            read_points_numpy(self.point_cloud_in),
+            [(1.0, 2.0, 0.0), (10.0, 20.0, 30.0)]))
+
+        # Copy current state of the original message for later check
         old_data = copy.deepcopy(self.point_cloud_in.data)
-        point_cloud_transformed = tf2_sensor_msgs.do_transform_cloud(
-            self.point_cloud_in, self.transform_translate_xyz_300)
-
-        k = 300
-        expected_coordinates = [(1+k, 2+k, 0+k), (10+k, 20+k, 30+k)]
-        new_points = list(point_cloud2.read_points(point_cloud_transformed))
-        print('new_points are %s' % new_points)
-        assert(expected_coordinates == new_points)
+
+        # Apply transform
+        point_cloud_transformed = do_transform_cloud(
+            self.point_cloud_in, transform_translate_xyz)
+
+        # Expected output
+        expected_coordinates = self.points + k
+
+        new_points = read_points_numpy(point_cloud_transformed)
+        self.assertTrue(np.allclose(expected_coordinates, new_points))
+
         # checking no modification in input cloud
-        assert(old_data == self.point_cloud_in.data)
+        self.assertEqual(old_data, self.point_cloud_in.data)
+
+    def test_simple_rotation_transform(self):
+        # Create a transform containing only a 180° rotation around the x-axis
+        transform_rot = TransformStamped()
+        transform_rot.transform.rotation.x = 1.0
+        transform_rot.transform.rotation.w = 6.123234e-17
```

Comment:
This can just be set to zero right? Same remark for line 104 and 123.

```suggestion
        transform_rot.transform.rotation.w = 0
```